### PR TITLE
Create user with automatic credentials and email

### PR DESCRIPTION
**Pull request by alfredoct96**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | https://github.com/mautic/mautic-documentation/pull/181
| Related developer documentation PR URL | N/A
| Issue(s) addressed                     | N/A

#### Description:

Added new method to create a new user.

I added a checkbox  "Generate random password and send an email with credentials to user. ".

If it is checked, inputs password will be hidden and the password will be automatically generated, 
in addition, an email with the credentials will be sent to the user.

newusermautic

#### Steps to test this PR:
1. Checkout to the branch of this PR
2. Generate new assets with `php bin/console mautic:assets:generate` command and clear cache with `php bin/console cache:clear` command
3. Configure email settings
4. Go to new user page
5. Check "Generate random password and send an email with credentials to user. "
6. Wait for email

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/9733"></a>

### CODE REVIEW

**Comment by dennisameling** [comment]
Hi @alfredoct96! Thanks for your PR 🚀 we just upgraded Mautic to Symfony 4 and this PR will need to be updated. Since this PR has conflicts now, could you please do a `git merge upstream/features` locally to resolve the conflicts? Thanks in advance!

**Comment by alfredoct96** [comment]
> Hi @alfredoct96! Thanks for your PR  we just upgraded Mautic to Symfony 4 and this PR will need to be updated. Since this PR has conflicts now, could you please do a `git merge upstream/features` locally to resolve the conflicts? Thanks in advance!

Done @dennisameling

File: app/bundles/UserBundle/Controller/UserController.php [hunk 1]
```diff
@@ -136,10 +136,15 @@ public function newAction()
             if (!$cancelled = $this->isFormCancelled($form)) {
                 //check to see if the password needs to be rehashed
                 $formUser          = $this->request->request->get('user', []);
+                $automaticPassword = $formUser['automaticPassword'];
+
+                error_log('stampo il type di auto');
+                error_log(gettype($automaticPassword));
+                error_log($automaticPassword);
```
Comment (kuzmany) [inline app/bundles/UserBundle/Controller/UserController.php]: These debug lines should not be part of this PR

**Review by kuzmany — DISMISSED** [review]

File: app/bundles/UserBundle/Controller/UserController.php (hunk 1 above)
Comment (alfredoct96) [reply app/bundles/UserBundle/Controller/UserController.php]: My distraction, i deleted these lines

**Review by RCheesley — CHANGES_REQUESTED** [review]
Great PR, a couple of suggestions:

1. When selecting 'yes' the slider should turn green and be visually activated but it doesn't seem to do that.
2. When you do select 'yes' to generate a random password, I would expect that the password fields would be blanked out, preventing the user from creating a password.

Screencast here: https://watch.screencastify.com/v/LGbIgpjyRLnivvV5QcYx (sorry for the background noise!)

Maybe we should even have the switch set to yes by default, and hide the password fields, unless someone clicks the slider to 'no' at which point we would show the password field.

Pinging @adiux @volha-pivavarchyk for UX input.

**Comment by adiux** [comment]
I agree with the changes proposed by @RCheesley. Sending the password should be the default behavior. And the fields should be hidden (as in the PrintScreen). Great PR 💪

**Comment by RCheesley** [comment]
@alfredoct96 would you be able to take a look at the feedback in this PR?

**Comment by alfredoct96** [comment]
> lt, and hid

Hi @RCheesley !
I think you have this problem because you have not generated the new assets, you need to generate new assets and clear cache.
I modified the behavior and now by default is selected automatic password option.

**Comment by RCheesley** [comment]
> I think you have this problem because you have not generated the new assets, you need to generate new assets and clear cache.

I don't think that is the case as we always work in dev mode when testing with Gitpod but I will check it again! Thanks!

File: app/bundles/EmailBundle/Translations/en_US/messages.ini [hunk 2]
```diff
@@ -461,3 +461,5 @@ mautic.email.stat.tooltip="Details may not match summary numbers if the contact
 mautic.email.stat.simple.tooltip="Details may not match summary numbers if the contact no longer exists in your Mautic Account"
 mautic.email.associated.contacts="Contacts"
 mautic.email.segment.choice.generated_email_domain="Email Domain"
+mautic.email.user.new_user.subject="Your username and password"
+mautic.email.user.new_user.body="Username: %username% <br />Password: %password% <br />Click <a href='%link%'>here</a> to login page"
```
Comment (RCheesley) [inline app/bundles/EmailBundle/Translations/en_US/messages.ini]:
```suggestion
mautic.email.user.new_user.body="A new account has been set up for you in Mautic. Use the credentials below to log in.<br /><br />Username: %username% <br />Password: %password% <br /><br />Click <a href='%link%'>here</a> to go to the login page. <br /><br />It is highly recommended that you set a new password and configure your signature when you first log in, located in your Account settings under your avatar in the top right."
```

I have added some line breaks as the email content was quite squashed, and also a bit more context so it is less likely to trigger spam-related alarm bells! 🚨 

I think that if we are sending the password in plain text when this email is sent, we should be responsible and inform the user that they are strongly recommended to set a password (and their signature) on login.

I don't know if we can force this but the least we could do is to inform them.

Also I noticed that we are just using the URL of the instance - what if someone has a custom page for that? I think we should append /s/login to the URL - what do you think @alfredoct96 ?

**Review by RCheesley — CHANGES_REQUESTED** [review]
Some changes to the email text to review and a ❓ about the link that is being used, otherwise works great - set to yes by default, and fields for password are shown if no is selected.

File: app/bundles/EmailBundle/Translations/en_US/messages.ini (hunk 2 above)
Comment (alfredoct96) [reply app/bundles/EmailBundle/Translations/en_US/messages.ini]: Done

**Comment by RCheesley** [comment]
OK, let's try and get this merged into 5.0 alpha. @alfredoct96 can you rebase to 5.x and then we can get this prioritised to be tested and reviewed on Friday in our Open Source Friday sprint. Sorry it's taken _so darn long_ to get to this point! Let's get it over the line! 🚀

**Comment by RCheesley** [comment]
Hey @alfredoct96 please can you rebase this to 5.x - would ❤️ to get this merged in time for the 5.0 alpha release!

**Comment by alfredoct96** [comment]
> OK, let's try and get this merged into 5.0 alpha. @alfredoct96 can you rebase to 5.x and then we can get this prioritised to be tested and reviewed on Friday in our Open Source Friday sprint. Sorry it's taken _so darn long_ to get to this point! Let's get it over the line! rocket

Hi @RCheesley,
doing this rebase is very complex due to the various conflicts and the risk of error is high, my advice is to recreate another branch starting from 5.x and cherry pick these commits, what do you say?

**Comment by kuzmany** [comment]
@alfredoct96 choose the fastest way for you. Rebase is just option I used to use different way as resolve so many conflicts with rebase mechanism

**Comment by alfredoct96** [comment]
> @alfredoct96 choose the fastest way for you. Rebase is just option I used to use different way as resolve so many conflicts with rebase mechanism

I did this pr https://github.com/mautic/mautic/pull/11729

**Comment by RCheesley** [comment]
OK - closing this PR in favour of #11729 thanks @alfredoct96 🚀
